File: frontend/src/services/warehouseService.ts
import axiosInstance from '../config/axios';
import type { PurchaseOrderResponse } from './purchaseOrderService';

export interface ReceiveItemRequest {
  purchaseOrderItemId: string;
  quantityReceived: number;
  notes?: string;
}

export interface ReceivePurchaseOrderRequest {
  actualArrivalDate?: string;
  items: ReceiveItemRequest[];
}

export const warehouseService = {
  inbound: async (): Promise<PurchaseOrderResponse[]> => {
    const response = await axiosInstance.get('/api/warehouse/inbound');
    return response.data;
  },
  receive: async (purchaseOrderId: string, data: ReceivePurchaseOrderRequest): Promise<PurchaseOrderResponse> => {
    const response = await axiosInstance.post(`/api/warehouse/purchase-orders/${purchaseOrderId}/receive`, data);
    return response.data;
  },
};

File: backend/src/main/java/com/ooas/service/WarehouseService.java
package com.ooas.service;

import com.ooas.dto.PurchaseOrderResponse;
import com.ooas.dto.ReceivePurchaseOrderRequest;
import java.util.List;

public interface WarehouseService {
    List<PurchaseOrderResponse> inbound();
    PurchaseOrderResponse receive(String purchaseOrderId, ReceivePurchaseOrderRequest request);
}
